judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/updater/ContestScoreboardUpdater.java

```diff
@@ -0,0 +1,131 @@
+package judgels.uriel.contest.scoreboard.updater;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.dropwizard.hibernate.UnitOfWork;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import judgels.sandalphon.api.submission.Submission;
+import judgels.uriel.api.contest.Contest;
+import judgels.uriel.api.contest.module.ContestModulesConfig;
+import judgels.uriel.api.contest.scoreboard.ContestScoreboardData;
+import judgels.uriel.api.contest.scoreboard.ContestScoreboardType;
+import judgels.uriel.api.contest.scoreboard.ScoreboardState;
+import judgels.uriel.contest.contestant.ContestContestantStore;
+import judgels.uriel.contest.module.ContestModuleStore;
+import judgels.uriel.contest.problem.ContestProblemStore;
+import judgels.uriel.contest.scoreboard.ContestScoreboardStore;
+import judgels.uriel.contest.scoreboard.ScoreboardProcessorRegistry;
+import judgels.uriel.contest.submission.ContestSubmissionStore;
+
+public class ContestScoreboardUpdater implements Runnable, Cloneable {
```

Comment:
So ideally, this should not know at all about it being a `Runnable`.

(1) it should not have the mutable `contest` field
(2) `run()` -> e.g. `updateScoreboard(Contest contest)`
(3) delete the non-standard pattern of cloning this just for the sake of injecting new contest :)

Then, in the dispatcher, you can now simply dispatch the task `() -> scoreboardUpdater.updateScoreboard(contest)`.